### PR TITLE
Refactor require-ing for supported analyzers

### PR DESCRIPTION
**Pull request by ngan**
From the discussion in https://github.com/igorkasyanchuk/active_storage_validations/pull/326, I thought it'd be a fun exercise to see what it would look like to extract support detection. This PR does the follow:
- Adds a new abstract `supported?` method that allows analyzers to do whatever to determine whether or not it's supported. This usually means loading the library and catching the load error.
- The support list is held (and cached) in a class attribute (Hash) for quick, subsequent lookups.
- This also changes `require "whatever" unless defined?(Whatever)` to just `require "whatever"` since it safeguards against things that happen to define `Whatever` without loading the library/gem.

### CODE REVIEW

File: lib/active_storage_validations/analyzer/image_analyzer/vips.rb [hunk 1]
```diff
@@ -47,6 +40,13 @@ def image_from_path(path)
       end
     end
 
+    def supported?
+      require "vips"
+      true
+    rescue LoadError
+      false
```
Comment (Mth0158) [inline lib/active_storage_validations/analyzer/image_analyzer/vips.rb:47]: Why did you remove the logger.info "..." line? Plus you left in mini_magick analyzer

**Review by Mth0158 — CHANGES_REQUESTED** [review]
Hi again @ngan 
Thanks for this PR, that solves yesterday's point! It's ok to me (except the logger point ?).

I'll also run it through @igorkasyanchuk since he was part of the discussion :) @igorkasyanchuk let us know what you think of this

File: lib/active_storage_validations/analyzer/image_analyzer/vips.rb (hunk 1 above)
Comment (ngan) [reply lib/active_storage_validations/analyzer/image_analyzer/vips.rb:48]: Whoops! Accidental 😅

**Review by Mth0158 — APPROVED** [review]
